### ISSUE TITLE
[ERROR] Graph Must Contain Start Vertex

### ISSUE THREAD

**Issue by yasmin259** [secure-software-engineering/upcy#16]
Hello, I have a problem pretty much similar to the other issue raised previously, where while running the last command, it should give me the .csv with the results, but it throws the following error, apparently the error indicates that the graph must contain the start vertex. I have tried with multiple projects but still getting the same message each time. 
<img width="952" alt="1" src="https://github.com/user-attachments/assets/86d7f919-dbbd-4898-bb17-8195e3c51935">
<img width="959" alt="2" src="https://github.com/user-attachments/assets/e165146a-976a-44c6-9c2f-a844fb1ded9a">

**Comment by yasmin259** [comment]
Hi @anddann. 

Thank you for your response. I have attached the dependency graph and the pom.xml files for your reference.  The project is  not multi-module  and I was able to run the "mvn clean compile install" successfully. Thank you. 

[pom.zip](https://github.com/user-attachments/files/17333544/pom.zip)
 
[dependency-graph.json](https://github.com/user-attachments/files/17333546/dependency-graph.json)


<img width="947" alt="1" src="https://github.com/user-attachments/assets/175ea303-1893-4c9d-bf53-c4719b3e7925">
<img width="946" alt="2" src="https://github.com/user-attachments/assets/0b3a15fd-832c-4435-9010-3943932d7052">

**Comment by anddann** [comment]
Okay, the graph contains the project node. After running mvn install, does UpCy work?
I've not tested UpCy with war-files, however, it should rune fine.

**Comment by yasmin259** [comment]
I tried running with the "-module ./" but getting the same output and no .csv file created. 

<img width="953" alt="module" src="https://github.com/user-attachments/assets/9f6705d4-1bc9-4fc0-a114-0fc5683f5a9a">
<img width="953" alt="module2" src="https://github.com/user-attachments/assets/5547846e-1e39-4953-a4f0-96dc1413f412">

**Comment by anddann** [comment]
Okay, I'll try to reproduce the error with the given pom.xml

**Comment by yasmin259** [comment]
Thank you. 

One more question, I tried running UPCY on some projects from the experimental-results--dataset folder, the tool is running fine however its always showing "No solution found in Neo4j" as attached below.  what is the reason for that?

 
<img width="862" alt="pmml" src="https://github.com/user-attachments/assets/c6ead686-f1c3-4162-ae93-f1d48507a082">
<img width="856" alt="pmml0" src="https://github.com/user-attachments/assets/cee9423e-ccb3-4f6a-b5e6-99468d2cf807">
<img width="854" alt="pdfcompare" src="https://github.com/user-attachments/assets/d2765333-b0ad-421c-ae99-5f17a3e99636">
<img width="851" alt="pdfcompare2" src="https://github.com/user-attachments/assets/062b08e1-bff9-4c8b-8510-34dc61e1c5b6">

**Comment by anddann** [comment]
Actually, in the given dataset it is not always showing no solution found.
- for some projects the tool does not find a solution in Neo4j
- in multiple case UpCy tries multiple options and some of them do not yield a result, then it shows no solution found and then continues to search..

Your Neo4j database maybe empty? outdated or invalid?

**Comment by anddann** [comment]
@yasmin259 I've debugged your project.
The issue is

```
<dependency>
			<groupId>javax.servlet</groupId>
			<artifactId>javax.servlet-api</artifactId>
			<version>3.1.0</version>
			<scope>provided</scope>
</dependency>
```

in particular the `<scope>provided</scope>` as written in the paper, UpCy only checks for compile dependencies.
Since  "[provided] indicates you expect the JDK or a container to provide the dependency at runtime" 
https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html, we excluded those as the dependency that is used depends on the runtime environment and may be updated/exchanged there.

However, the error message is misleading. I'll improve the message.
